### PR TITLE
Fix: Bug #539 - network.py crashes with ZeroDivisionError on some systems

### PR DESCRIPTION
**Pull request by rodwsmith**
Fix: Bug #539 - network.py crashes with ZeroDivisionError on some systems

## Description

This PR adds protection against a divide-by-zero error, causing the script to not perform the division (actually a modulo operator) or try to use the iperf3 option that relies on the result of that operator. This should be OK, since the relevant code was added to support machines with multiple NUMA nodes, and the affected system seems to lack these, so omitting the NUMA-node option should be harmless.

## Resolved issues

Fixes: https://github.com/canonical/checkbox/issues/539
Resolves: https://warthogs.atlassian.net/browse/SERVCERT-1165

## Documentation

No documentation changes needed; this just fixes a bug.

## Tests

I was unable to re-create the problem, since I lack a Raspberry Pi; however, I faked it by temporarily adjusting the code. I tested the fix against this adjusted code, and it worked; the divide-by-zero error was bypassed and the script ran normally. I then reverted the adjustment to fake the problem and left the fix in place. It's just a one-line change.

### CODE REVIEW

File: providers/base/bin/network.py [hunk 1]
```diff
@@ -288,7 +288,7 @@ def run(self):
         t = []
         results.clear()
         for thread_num in range(0, python_threads):
-            if self.iperf3:
+            if self.iperf3 and len(core_list) > 0:
```
Comment (kissiel) [inline providers/base/bin/network.py:291]: Note: Checking if len(core_list) > 0 is essentially the same as checking if core_list. In Python, empty collections are considered False when converted to a boolean context, so you can simplify your condition check. Of course it's an extremely minor thing, and the PR is great as is!

**Review by kissiel — APPROVED** [review]
TYVM!
One comment below.

**Comment by baconYao** [comment]
Hi @rodwsmith,

I can do testing without any script crash problem after applying this fix.

The following attachment is my sideload result on RPI4 Jammy Desktop Image.
[submission_2023-06-13T01.39.31.106725.zip](https://github.com/canonical/checkbox/files/11730361/submission_2023-06-13T01.39.31.106725.zip)

Conclusion: LTGM

**Comment by rodwsmith** [comment]
> Hi @rodwsmith,
> 
> I can do testing without any script crash problem after applying this fix.

> Conclusion: LTGM

OK. Can you please approve the PR, then? The merge is blocked on that approval. Thanks!

**Comment by baconYao** [comment]
> > Hi @rodwsmith,
> > I can do testing without any script crash problem after applying this fix.
> 
> > Conclusion: LTGM
> 
> OK. Can you please approve the PR, then? The merge is blocked on that approval. Thanks!

@rodwsmith I don't have permission to merge it.

Hi @kissiel @pieqq, could you please help to merge this PR ? Thanks.

**Comment by kissiel** [comment]
Definitely!

**Comment by rodwsmith** [comment]
Thanks!

**Pull request merged**